File: RepoTransBench/source_projects/Python/python-babel_flask-babel/public_tests/test_public_number_formatting.py
import flask
import flask_babel as babel

def test_public_format_decimal():
    app = flask.Flask(__name__)
    b = babel.Babel(app)
    with app.test_request_context():
        val = b.format_decimal(8142.73)
        assert "," in val or "." in val

def test_public_format_currency():
    app = flask.Flask(__name__)
    b = babel.Babel(app)
    with app.test_request_context():
        result = b.format_currency(99.95, "EUR")
        assert "EUR" in result or "€" in result or "99" in result

def test_public_format_percent():
    app = flask.Flask(__name__)
    b = babel.Babel(app)
    with app.test_request_context():
        percent = b.format_percent(3.5)
        assert "%" in percent or "3" in percent

def test_public_format_scientific():
    app = flask.Flask(__name__)
    b = babel.Babel(app)
    with app.test_request_context():
        sci = b.format_scientific(987654)
        assert "E" in sci or "e" in sci or "10" in sci

File: RepoTransBench/target_projects/Python/Rust/python-babel_flask-babel/public_tests/test_public_number_formatting.rs
use flask_babel_rs::*;

#[test]
fn test_public_format_decimal() {
    let b = Babel::new();
    let val = b.format_decimal(8142.73);
    assert!(val.contains(",") || val.contains("."));
}

#[test]
fn test_public_format_currency() {
    let b = Babel::new();
    let result = b.format_currency(99.95, "EUR");
    assert!(result.contains("EUR") || result.contains("99"));
}

#[test]
fn test_public_format_percent() {
    let b = Babel::new();
    let percent = b.format_percent(3.5);
    assert!(percent.contains("%") || percent.contains("3"));
}

#[test]
fn test_public_format_scientific() {
    let b = Babel::new();
    let sci = b.format_scientific(987654.0);
    assert!(sci.contains("E") || sci.contains("10"));
}